recipes-kernel/linux/files/0023-ethernet-Swap-eth0-and-eth1-port.patch

```diff
@@ -0,0 +1,69 @@
+From 8dc174b64873a9be4a30862c1c952f460f2ef38d Mon Sep 17 00:00:00 2001
+From: Chao Zeng <chao.zeng@siemens.com>
+Date: Wed, 17 Jun 2020 13:20:16 +0800
+Subject: [PATCH 23/23] ethernet: Swap eth0 and eth1 port 	1. add the
+ alias support for the ethernet driver         2. modify the alias
+ configuration to swap the ethernet port
```

Comment:
Should be separated into two patches. The alias support needs to go to TI so that they integrate it into the driver (until that is upstream).

Reply:
@jan-kiszka  do you think if it's better to change the network interface name from user space, such as udev?

Reply:
It would be an alternative, indeed.

Looking at the kernel, there are a few drivers doing aliases:
```
drivers/net/ethernet/broadcom/genet/bcmmii.c:           id = of_alias_get_id(dn, "eth");
drivers/net/ethernet/samsung/sxgbe/sxgbe_platform.c:    plat->bus_id = of_alias_get_id(np, "ethernet");
drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c:  plat->bus_id = of_alias_get_id(np, "ethernet");
```

Reply:
Thinking about this further: an udev rule would allow to call the interfaces exactly like the ports (x1p1, x1p2).

Reply:
The concern is that if we add the alias support for the driver. In the future, we submit the patch to the upstream.  Anyone else uses this driver would cause some error. 
Maybe Ethernet  port conflict  if we assign the port with duplicate port number.  

Maybe use the Udev is a good alternative solution for this feature.